mypy/checker.py

```diff
@@ -654,6 +659,16 @@ def _visit_overloaded_func_def(self, defn: OverloadedFuncDef) -> None:
             self.fail(message_registry.INCONSISTENT_ABSTRACT_OVERLOAD, defn)
         if defn.impl:
             defn.impl.accept(self)
+            if (
+                isinstance(impl := defn.impl, Decorator)
```

Comment:
I feel this is too many walruses, especially with the `ct` variable name being reused.

Reply:
I agree. Only one walrus left.